sdc/tests/test_series.py

```diff
@@ -5382,158 +5379,119 @@ def test_impl(S, val):
             result = hpat_func(S2, value)
-            pd.testing.assert_series_equal(result_ref, result)
+            pd.testing.assert_series_equal(result, result_ref)
 
-    def test_series_setitem_for_slice_int(self):
-        def test_impl(S, val):
-            S[2:] = val
+    def test_series_setitem_idx_int_slice_all_dtypes(self):
+        def test_impl(S, idx, val):
+            S[idx] = val
             return S
-
         hpat_func = self.jit(test_impl)
-        data_to_test = [[0, 1, 2, 3, 4]]
 
-        for data in data_to_test:
-            S1 = pd.Series(data)
-            S2 = S1.copy(deep=True)
-            value = 50
-            result_ref = test_impl(S1, value)
-            result = hpat_func(S2, value)
-            pd.testing.assert_series_equal(result_ref, result)
-
-    def test_series_setitem_for_slice_float(self):
-        def test_impl(S, val):
-            S[2:] = val
-            return S
-
-        hpat_func = self.jit(test_impl)
-        data_to_test = [[0, 0, 0, np.nan, np.nan, 0, 0, np.nan, np.inf, 0, 0, np.inf, np.inf],
+        dtype_to_data = {
+            'int' :    [[0, 1, 2, 3, 4]],
+            'float' :  [[0, 0, 0, np.nan, np.nan, 0, 0, np.nan, np.inf, 0, 0, np.inf, np.inf],
                         [1.1, 0.3, np.nan, 1, np.inf, 0, 1.1, np.nan, 2.2, np.inf, 2, 2],
-                        [1, 2, 3, 4, np.nan, np.inf, 0, 0, np.nan, np.nan]]
-
-        for data in data_to_test:
-            S1 = pd.Series(data)
-            S2 = S1.copy(deep=True)
-            value = np.nan
-            result_ref = test_impl(S1, value)
-            result = hpat_func(S2, value)
-            pd.testing.assert_series_equal(result_ref, result)
-
-    @skip_numba_jit('NOT WORK, PROBLEM WITH StringArrayType in _str_ext.cpp')
-    @skip_sdc_jit
-    def test_series_setitem_for_slice_string(self):
-        def test_impl(S, val):
-            S[2:] = val
-            return S
-
-        hpat_func = self.jit(test_impl)
-        data_to_test = [['a', '', 'a', '', 'b', None, 'a', '', None, 'b'],
+                        [1, 2, 3, 4, np.nan, np.inf, 0, 0, np.nan, np.nan]],
+            'string' :  [['a', '', 'a', '', 'b', None, 'a', '', None, 'b'],
                         ['dog', None, 'NaN', '', 'cat', None, 'cat', None, 'dog', ''],
                         ['dog', 'NaN', '', 'cat', 'cat', 'dog', '']]
-
-        for data in data_to_test:
-            S1 = pd.Series(data)
-            S2 = S1.copy(deep=True)
-            value = 'Hello, world!'
-            result_ref = test_impl(S1, value)
-            result = hpat_func(S2, value)
-            pd.testing.assert_series_equal(result_ref, result)
-
-    def test_series_setitem_for_series_int(self):
-        def test_impl(S, ind, val):
-            S[ind] = val
+        }
+        dtype_to_values = {
+            'int' :    50,
+            'float' :  np.nan,
+            'string' : 'bird'
+        }
+
+        idx = slice(2, None)
+        for dtype in dtype_to_data:
```

Comment:
Let's use method `items()` of `dict` here in other tests below because of using `dict[key]`.

Reply:
Changed here and in the test below.